components/infobox/commons/infobox_widget_injector.lua

```diff
@@ -9,7 +9,14 @@
 local Class = require('Module:Class')
 
 ---@class WidgetInjector
-local Injector = Class.new()
+---@operator call(table?): WidgetInjector
+---@field parent table?
+local Injector = Class.new(
+	---@param self self
+	---@param parent table?
+	function(self, parent)
+		self.parent = parent
+end)
```

Comment:
Maybe something like this is a better naming, since it is not necessarily a parent? (Not entirely sure that caller is really a better name tho)
```suggestion
	function(self, caller)
		self.caller = caller
end)
```

Reply:
used it for now, if there are other/better name suggestions let me know